src/client/app/components/ChartDataSelectComponent.tsx

```diff
@@ -2,75 +2,90 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import React from 'react';
+import * as React from 'react';
 import MultiSelectComponent from './MultiSelectComponent';
+import { SelectOption } from '../types/items';
+import { fetchGroupsDetailsIfNeeded } from 'actions/groups';
+
+interface ChartDataSelectProps {
+	meters: SelectOption[];
+	groups: SelectOption[];
+	selectedMeters: SelectOption[];
+	selectedGroups: SelectOption[];
+	fetchMetersDetailsIfNeeded(): Promise<any>;
+	fetchGroupsDetailsIfNeeded(): Promise<any>;
+	selectMeters(meterIDs: number[]): Promise<any>;
+	selectGroups(groupIDs: number[]): Promise<any>;
+}
 
 /**
  * A component which allows the user to select which data should be displayed on the chart.
  */
-export default class ChartDataSelectComponent extends React.Component {
-	constructor(props) {
+export default class ChartDataSelectComponent extends React.Component<ChartDataSelectProps, {}> {
+	constructor(props: ChartDataSelectProps) {
 		super(props);
 		this.handleMeterSelect = this.handleMeterSelect.bind(this);
 		this.handleGroupSelect = this.handleGroupSelect.bind(this);
 	}
 
 	/**
- 	 * Called when this component mounts
- 	 * Dispatches a Redux action to fetch meter information
- 	 */
-	componentWillMount() {
+	 * Called when the component mounts.
+	 * Fetch all meter and group info, for display.
+	 */
+	public componentWillMount() {
 		this.props.fetchMetersDetailsIfNeeded();
 		this.props.fetchGroupsDetailsIfNeeded();
 	}
 
-	/**
-	 * Handles a change in meter selection
-	 * @param {Object[]} selection An array of {label: string, value: {type: string, id: int}} representing the current selection
-	 */
-	handleMeterSelect(selection) {
-		this.props.selectMeters(selection.map(s => s.value));
-	}
-
-	/**
-	 * Handles a change in group selection
-	 * @param {Object[]} selection An array of {label: string, value: {type: string, id: int}} representing the current selection
-	 */
-	handleGroupSelect(selection) {
-		this.props.selectGroups(selection.map(s => s.value));
-	}
-
-	render() {
-		const divBottomPadding = {
+	public render() {
```

Comment:
Recommend always having the render() method be at the bottom of the class, just a convention.

Reply:
It's a linter rule.